src/component/contents/DraftEditorContents.react.js

```diff
@@ -197,21 +199,80 @@ class DraftEditorContents extends React.Component {
         wrapperTemplate,
         key,
         offsetKey,
+        draftWrapperTemplate: configForType.wrapper,
+        customWrapperTemplate: customWrapperTemplate,
       });
 
-      if (wrapperTemplate) {
+      if (draftWrapperTemplate || customWrapperTemplate) {
         currentDepth = block.getDepth();
       } else {
         currentDepth = null;
       }
       lastWrapperTemplate = wrapperTemplate;
+      lastCustomWrapperTemplate = customWrapperTemplate;
     }
 
-    // Group contiguous runs of blocks that have the same wrapperTemplate
+    // Group contigious runs of blocks
     const outputBlocks = [];
-    for (let ii = 0; ii < processedBlocks.length; ) {
+    for (let ii = 0; ii < processedBlocks.length;) {
       const info = processedBlocks[ii];
-      if (info.wrapperTemplate) {
+
+      // Group by customWrapperTemplate (if exists)
+      // Note: If we ever want to remove customWrapperTemplates, remove this if block
+      if (info.customWrapperTemplate) {
+        // Group by custom template
+        const blocksGroupedByCustomTemplate = [];
+        do {
+          blocksGroupedByCustomTemplate.push(processedBlocks[ii]);
+          ii++;
+        } while (
+          ii < processedBlocks.length &&
+          processedBlocks[ii].customWrapperTemplate == info.customWrapperTemplate
+        );
+
+        // After grouping by customWrapperTemplate, 
+        // group those blocks by draftjs wrapper template
```

Comment:
As a minor comment: since this is Draft it sounds odd to refer to Draft as if it were a 3rd party

Reply:
How about internalWrapper ? Naming things is hard, haha